File: 240428/전부 겹치는 선분/overlapping-line-segments.py
N = int(input())
lines = [tuple(map(int, input().split())) for _ in range(N)]

total_line = [0] * 101

for x, y in lines:
    for i in range(x, y + 1):
        total_line[i] += 1

if N in total_line:
    print("Yes")
else:
    print("No")

File: 260403/전부 겹치는 선분/overlapping-line-segments.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N;
    cin >> N;
    vector<pair<int, int>> lines(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> lines[i].first >> lines[i].second;
    }

    vector<int> total_line(101, 0);

    for (auto &line : lines) 
    {
        int x = line.first;
        int y = line.second;
        for (int i = x; i <= y; i++) 
        {
            total_line[i] += 1;
        }
    }

    bool found = false;
    for (int i = 0; i < 101; i++) 
    {
        if (total_line[i] == N) 
        {
            found = true;
            break;
        }
    }

    if (found) 
    {
        cout << "Yes\n";
    } else {
        cout << "No\n";
    }
    return 0;
}